### PR TITLE
Conditionally install RPi.GPIO and spidev only on linux targets (not OSX/Darwin)

### PR DESCRIPTION
**Pull request by rm-hull**
This goes some way to remediate the issue described in https://github.com/rm-hull/luma.emulator/issues/18

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -24,6 +24,16 @@ def read_file(fname):
     "pytest-cov"
 ]
 
+install_deps = [
+    'pillow>=4.0.0',
+    'smbus2',
+    'monotonic;python_version<"3.3"'
+]
+
+if os.uname()[0] != 'Darwin':
```
Comment (thijstriemstra) [inline setup.py]: @rm-hull can you use ``platform.system()`` instead?

Comment (thijstriemstra) [reply setup.py]: And add a changelog entry.

Comment (thijstriemstra) [reply setup.py]: and maybe use extend instead e.g.:

```python
rpi_deps = ['spidev', 'RPI.GPIO']
install_deps.extend(rpi_deps)
```

**Comment by thijstriemstra** [comment]
And I'm also [seeing](https://travis-ci.org/rm-hull/luma.core/jobs/271852801) this traceback on Python 3.6:

```
Exception:

Traceback (most recent call last):

  File "/home/travis/virtualenv/python3.6.2/lib/python3.6/site-packages/pip/basecommand.py", line 215, in main

    status = self.run(options, args)

  File "/home/travis/virtualenv/python3.6.2/lib/python3.6/site-packages/pip/commands/install.py", line 342, in run

    prefix=options.prefix_path,

  File "/home/travis/virtualenv/python3.6.2/lib/python3.6/site-packages/pip/req/req_set.py", line 784, in install

    **kwargs

  File "/home/travis/virtualenv/python3.6.2/lib/python3.6/site-packages/pip/req/req_install.py", line 851, in install

    self.move_wheel_files(self.source_dir, root=root, prefix=prefix)

  File "/home/travis/virtualenv/python3.6.2/lib/python3.6/site-packages/pip/req/req_install.py", line 1064, in move_wheel_files

    isolated=self.isolated,

  File "/home/travis/virtualenv/python3.6.2/lib/python3.6/site-packages/pip/wheel.py", line 247, in move_wheel_files

    prefix=prefix,

  File "/home/travis/virtualenv/python3.6.2/lib/python3.6/site-packages/pip/locations.py", line 140, in distutils_scheme

    d = Distribution(dist_args)

  File "/home/travis/virtualenv/python3.6.2/lib/python3.6/site-packages/setuptools/dist.py", line 365, in __init__

    self._finalize_requires()

  File "/home/travis/virtualenv/python3.6.2/lib/python3.6/site-packages/setuptools/dist.py", line 372, in _finalize_requires

    if not self.install_requires:

AttributeError: 'Distribution' object has no attribute 'install_requires'
```

**Comment by rm-hull** [comment]
Its happening on master too - just documented it in #98

File: .travis.yml [hunk 2]
```diff
@@ -17,7 +17,7 @@ matrix:
         - python: 2.7
           env: TOXENV=qa
 
-install: pip install -U setuptools pip tox coveralls
+install: pip install -IU setuptools pip tox coveralls
```
Comment (thijstriemstra) [inline .travis.yml]: whats `I`?

Comment (thijstriemstra) [reply .travis.yml]: maybe use long versions instead, e.g. `--ignore-installed` and `--upgrade` @rm-hull

**Comment by thijstriemstra** [comment]
Also, it shouldn't install on Windows either @rm-hull.

File: CHANGES.rst [hunk 3]
```diff
@@ -4,6 +4,8 @@ ChangeLog
 +------------+---------------------------------------------------------------------+------------+
 | Version    | Description                                                         | Date       |
 +============+=====================================================================+============+
+| *Upcoming* | * Don't install RPi.GPIO & spidev if setup running on OSX           | 2017/09/04 |
```
Comment (thijstriemstra) [inline CHANGES.rst]: @rm-hull update changelog and merge, thanks.

**Pull request merged**